client/asset/eth/rpcclient.go

```diff
@@ -256,6 +258,56 @@ func (c *rpcclient) initiate(txOpts *bind.TransactOpts, netID int64, refundTimes
 	return c.es.Initiate(txOpts, big.NewInt(refundTimestamp), secretHash, *participant)
 }
 
+// initiateGas checks the amount of gas that is used for a call to the initiate function.
+func (c *rpcclient) initiateGas(ctx context.Context, refundTimestamp int64, secretHash [32]byte, participant *common.Address, contractAddress *common.Address) (uint64, error) {
+	parsedAbi, err := abi.JSON(strings.NewReader(swap.ETHSwapABI))
+	if err != nil {
+		return 0, err
+	}
+	data, err := parsedAbi.Pack("initiate", big.NewInt(refundTimestamp), secretHash, participant)
```

Comment:
I was thinking to do that also at first too, but then I thought it would be nicer if this function was more flexible and took the arguments that the initiate function on the contract takes and returned the amount of gas it would use with those arguments. If someone initiated a swap where the `secretHash` was 1, then this estimation would constantly break. If you feel strongly about it I'm fine with changing it though, as that probably would never happen.

Reply:
I might be misunderstanding this conversation, but `[32]byte{0x01}` is still a 32-byte array, it's just a syntax for setting one byte of that array so as to make it not the zero value for the secret hash.